app/components/windows/WidgetEditor.vue.ts

```diff
@@ -156,7 +162,20 @@ export default class WidgetEditor extends Vue {
   }
 
   toggleCustomCode(enabled: boolean) {
-    const newSettings = { ...this.wData.settings, custom_enabled: enabled };
-    this.widget.getSettingsService().saveSettings(newSettings)
+    let newSettings = { ...this.wData.settings, custom_enabled: enabled };
+    if (this.selectedVariation) {
+      newSettings = { ...this.wData.settings };
+      Object.keys(newSettings).forEach((type) => {
+        const variations = newSettings[type].variations;
+        const found = variations && variations.find(
+          (vari: IAlertBoxVariation) => this.selectedVariation.id === vari.id
+        );
+        if (found) {
+          found.settings.customHtmlEnabled = true;
+        }
+      });
+    }
+    console.log(newSettings);
+    this.widget.getSettingsService().saveSettings(newSettings);
```

Comment:
This is a pretty inelegant solution but i'm not sure of a better way to do it while not passing in like 100 props.

Reply:
It worth to move this method to `widget-settings` service and create an overridden version of it for `alertbox` service